Single table inheritance

GonSanchezS opened the pull request:
To start with single table inheritance, I created a pawn model that inherits all the `Piece` attributes. I also wrote a test to check if a pawn is, in fact, a pawn.

GonSanchezS commented:
I'd love to know everyone's thoughts on this. Once it's greenlighted, I'll moved forward with the rest of the pieces.

lasley commented on db/migrate/20161119163703_create_pawns.rb:
```diff
@@ -0,0 +1,8 @@
+class CreatePawns < ActiveRecord::Migration
+  def change
+    create_table :pawns do |t|
```
Single table inheritance means one table. You're creating a second here

lasley commented:
* http://api.rubyonrails.org/classes/ActiveRecord/Inheritance.html
* https://eewang.github.io/blog/2013/03/12/how-and-when-to-use-single-table-inheritance-in-rails/
* https://samurails.com/tutorial/single-table-inheritance-with-rails-4-part-1/

GonSanchezS commented:
Crap. That was careless. I didn't generate a migration, but it was created
when running 'rails g model'. I'll fix this in a bit.

El El sáb, 19 de nov. de 2016 a las 14:22, Dave Lasley <
notifications@github.com> escribió:

> *@lasley* commented on this pull request.
> ------------------------------
>
> In db/migrate/20161119163703_create_pawns.rb
> <https://github.com/the-cheaters/up-all-knight/pull/14#pullrequestreview-9348970>
> :
>
> > @@ -0,0 +1,8 @@
> +class CreatePawns < ActiveRecord::Migration
> +  def change
> +    create_table :pawns do |t|
>
> Single table inheritance means one table. You're creating a second here
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/the-cheaters/up-all-knight/pull/14#pullrequestreview-9348970>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AM2UTgiDUkgirb4j02etbTuciB3Ev6LPks5q_zBRgaJpZM4K3TsS>
> .
>

GonSanchezS commented:
OK, I reverted back to the initial state of the branch, and create the `Pawn` model using the instructions on Railcast. I added the corresponding nested factories, but the test isn't passing. The error is:

_ActiveRecord::RecordInvalid:
Validation failed: Email can't be blank, Password can't be blank._

I'll be back to this later today, but any thoughts are appreciated.

kellylougheed commented:
Hey @GonSanchezS, I ran into the same error with email & password. Basically, for your tests, you have to make sure the piece has a player assigned to it (with an email and password). I created two players, a game, and made sure each piece had a game and a player. You can see how I did the tests for piece_spec.rb here: https://github.com/the-cheaters/up-all-knight/pull/13/commits/926bc2c4b78e724a7058da68f7d81bee0a5c69a8#diff-323325a24dc7c5853deae0cb726c158c

Hope that helps!

GonSanchezS commented:
@kellylougheed thanks for the help! I used some of your code to get the test to pass. Feel free to take a look, and I can move forward with creating models for the rest of the pieces.

kellylougheed commented:
Looks good to me @GonSanchezS! It'll actually be super convenient that we have the exact same "let" statements at the top of piece_spec.rb when we merge branches.

lasley commented on spec/models/piece_spec.rb:
```diff
@@ -1,5 +1,17 @@
+
+  describe 'a Pawn' do
+    it 'should be a pawn' do
+      pawn = FactoryGirl.create(:pawn, x_position: 2, y_position: 1, game_id: game.id, player_id: black_player.id)
```
Why all these attributes in the factorygirl create, but no default attributes in the factory? These should all be moved to the factory IMO

GonSanchezS replied:
I moved them to the factory, but still had to assign a player in the factorygirl create. Otherwise, the test won't pass.

kellylougheed commented:
Looks good to me. I'll resolve the conflicts in master and merge it.
